pennylane/circuit_graph.py

```diff
@@ -469,9 +471,42 @@ def get_depth(self):
                 self._operation_graph = self._graph.subgraph(
                     list(self._graph.nodes().index(node) for node in self.operations)
                 )
+                self._extend_graph(self._operation_graph)
                 self._depth = rx.dag_longest_path_length(self._operation_graph) + 1
         return self._depth
 
+    def _extend_graph(self, graph: rx.PyDiGraph) -> rx.PyDiGraph:
```

Comment:
I have a bit of caution adding a (private) method to `circuit_graph`. Could we instead have this functionality live in `resource`? I.e., if our circuit contains a `OperationWithResources`, then we call a `_extend_graph` private function which updates the underlying `PyDiGraph` before calculating its depth?

Reply:
I can move it to be a private method in resources. I don't know if that fits with best practices. How about as an alternative, it's not a private method in the class, but rather a private function in the file?